### PR TITLE
15 - update metering API

### LINKED ISSUE #15: Update metering API
Currently metering API returns all aggregated data for continuous queries. 
We want to limit the amount of records fetched from given timestamp. For example:
```
/queries/5m?from=123456789
```

### PR DESCRIPTION
**Pull request by piotrkosecki**
resolves #15

### CODE REVIEW

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala [hunk 1]
```diff
@@ -27,22 +27,22 @@ object Models {
 trait InfluxDbRepo {
 
   /** Reads amount of queries grouped by 5minute intervals */
-  def get5minQueries(apiKeys: List[String]): Future[List[QueryCQ]]
+  def get5minQueries(apiKeys: List[String], from: Option[Long]): Future[List[QueryCQ]]
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala:30]: Can we have docs on all of the following parameters.
What is it representing, using which time unit, and relative to now?

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala [hunk 2]
```diff
@@ -52,15 +52,19 @@ class InfluxDbRepoImpl(cfg: InfluxDbConfig)(
...
 
   /** Reads amount of queries grouped by 5minute intervals */
-  override def get5minQueries(apiKeys: List[String]): Future[List[QueryCQ]] =
+  override def get5minQueries(apiKeys: List[String], from: Option[Long]): Future[List[QueryCQ]] =
     client
       .measurement[QueryCQ](cfg.database, "five_minute_queries_measurement")
       .read(
-        s"select time, apiKey, count from a_day.five_minute_queries_measurement where apiKey =~ /${apiKeys.mkString("|")}/"
+        s"select time, apiKey, count from a_day.five_minute_queries_measurement where apiKey =~ /${apiKeys.mkString("|")}/" + timeConstraint(
+              from
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala:66]: Is this appending a string fragment to an interpolation? Why not interpolate this value as well?

File: src/main/scala/tech/cryptonomic/nautilus/metering/routes/Routes.scala [hunk 3]
```diff
@@ -16,40 +16,42 @@ class Routes(influxDbRepo: InfluxDbRepo)(implicit system: ActorSystem, materiali
...
-      )
+            )
+          },
+          pathPrefix("routes") {
+            concat(
+              path("5m") {
+                complete(influxDbRepo.get5minRoute(keys.toList, from))
+              },
+              path("24h") {
+                parameters('apiKey.*) { keys =>
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/metering/routes/Routes.scala:37]: isn't this a repetition?

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala [hunk 4]
```diff
@@ -126,11 +134,13 @@ class InfluxDbRepoImpl(cfg: InfluxDbConfig)(
       }
 
   /** Reads amount of hits per IP grouped by 24hour intervals */
-  override def get24hIp(apiKeys: List[String]): Future[List[IpCQ]] =
+  override def get24hIp(apiKeys: List[String], from: Option[Long]): Future[List[IpCQ]] =
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala:137]: secondary remark: why aren't these methods named consistently with the  corresponding columns?

daily, fiveMinutes, ...

Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala]: ok, renamed

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala (hunk 2 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala]: ok, changed

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala (hunk 1 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala]: ok, I've added comments in the implementation

File: src/main/scala/tech/cryptonomic/nautilus/metering/routes/Routes.scala (hunk 3 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/metering/routes/Routes.scala]: yes, I must have missed that, changed

File: src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala [hunk 5]
```diff
@@ -83,12 +94,16 @@ class InfluxDbRepoImpl(cfg: InfluxDbConfig)(
         case Right(value) => value.toList
       }
 
-  /** Reads amount of hits per route grouped by 5minute intervals */
-  override def get5minRoute(apiKeys: List[String]): Future[List[RouteCQ]] =
+  /** Reads amount of hits per route grouped by 5minute intervals
+    * @param apiKeys list of API keys
+    * @param from optional timestamp in seconds
+    */
+  override def getFiveMinuesRoute(apiKeys: List[String], from: Option[Long]): Future[List[RouteCQ]] =
```
Comment (bhaskarkishore) [inline src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala:101]: Typo in function name, i.e `Minutes`

**Pull request merged**